src/v/cluster/tm_stm.cc

```diff
@@ -310,6 +310,9 @@ tm_stm::do_update_tx(tm_transaction tx, model::term_id term) {
 
     auto r = co_await replicate_quorum_ack(term, std::move(batch));
     if (!r) {
+        if (r.error() == raft::errc::shutting_down) {
+            co_return tm_stm::op_status::timeout;
```

Comment:
can't we just return a not_leader for the client to retry?  Returning a "timeout" during shutdown confused me.

Reply:
We can't use not_leader here, not_leader means "I can't perform operation because I'm not a leader"; shutting_down means that during an operation an error occurred and the true status of operation may be either passed or rejected. timeout has the same meaning, the operation timed out and its status is unknown